### PR TITLE
Zachmu/show indexes

### PR DESCRIPTION
**Pull request by zachmu**
Implemented new interface methods needed by latest go-mysql-server

### CODE REVIEW

File: go/libraries/doltcore/sqle/enginetest/dolt_harness.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func (d *doltHarness) NewTableAsOf(db sql.VersionedDatabase, name string, schema
 // branch with the given name.
 func (d *doltHarness) SnapshotTable(db sql.VersionedDatabase, name string, asOf interface{}) error {
 	ddb := db.(sqle.Database)
-	e := enginetest.NewEngineWithDbs(d.t, d.Parallelism(), []sql.Database{db}, nil)
+	e := enginetest.NewEngineWithDbs(d.t, d, []sql.Database{db}, nil)
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/sqle/enginetest/dolt_harness.go:123]: we don't need an index driver?

File: go/libraries/doltcore/sqle/dolt_index.go [hunk 2]
```diff
@@ -159,6 +160,21 @@ func (di *doltIndex) ID() string {
...
 }
 
+// IsUnique implements sql.Index
+func (di *doltIndex) IsUnique() bool {
+	return di.unique
+}
+
+// Comment implements sql.Index
+func (di *doltIndex) Comment() string {
+	return "" // TODO: store comments
```
Comment (Hydrocharged) [inline go/libraries/doltcore/sqle/dolt_index.go:170]: In `go/libraries/doltcore/sqle/tables::GetIndexes`, where we create the `doltIndex` objects, you have access to the index. Just like you're doing for `Unique` you can do for for `Comment` too.

File: go/libraries/doltcore/sqle/enginetest/dolt_harness.go (hunk 1 above)
Comment (zachmu) [reply go/libraries/doltcore/sqle/enginetest/dolt_harness.go:123]: Nope, got rid of that about a month ago. Tables provide their own indexes now, no driver needed.

File: go/libraries/doltcore/sqle/dolt_index.go (hunk 2 above)
Comment (zachmu) [reply go/libraries/doltcore/sqle/dolt_index.go:170]: Oh snap, I didn't think we had this! Added.

**Pull request merged**